analysis-module/src/main.py

```diff
@@ -34,16 +38,35 @@
     msg_propagation_times = analysis.compute_propagation_times(msgs_dict)
     msg_injection_times = analysis.compute_injection_times(injected_msgs_dict)
 
-    cpu_usage, memory_usage, bandwith_in, bandwith_out = prometheus.get_hardware_metrics(
-        topology_info,
-        min_tss,
-        max_tss, prom_port)
+    """ Generate stats depending on the type of measurements we have """
+    if infra_type == 'container-proc':
+        analysis_logger.G_LOGGER.info('Generating stats for container_proc infrastructure ...')
+
+        metrics_info, max_cpu_usage, max_memory_usage, total_network_usage, max_disk_usage, avg_samples_per_node = analysis_cproc.compute_process_level_metrics(simulation_path, config_obj)
+        
+        """ Build simulation summary """
+        summary = analysis_cproc.build_summary(config_obj, metrics_info, msgs_dict, node_logs, [], min_tss, max_tss, avg_samples_per_node)
+        
+        """ Generate Figure """
+        plotting.plot_figure_host_proc(msg_propagation_times, max_cpu_usage, max_memory_usage, total_network_usage, max_disk_usage, 
+                       msg_injection_times, summary['general'], summary['parameters'])
+        
+        """ Export summary """
+        analysis_cproc.export_sumary(simulation_path, summary)
+        
+    # Add cadvisor and host-proc logic cases bellow
+    else:
+
+        cpu_usage, memory_usage, bandwith_in, bandwith_out = prometheus.get_hardware_metrics(
+            topology_info,
+            min_tss,
+            max_tss, prom_port)
```

Comment:
I think already commented this in the main PR thread. Let's try to keep the main clean, and if any logic needs to be done, we should extract it into another location.